app/views/bus_stops/_form.haml

```diff
@@ -38,7 +38,7 @@
           %tr
             %td= f.label name
             - if options.kind_of?(Array)
-              %td= f.select name, options, include_blank: true
+              %td= f.select name, options |= [stop.send(name)], include_blank: true
```

Comment:
Interesting. Didn't know `|=` was a thing.

That being said, how feasible would it be to construct this in the controller? Or perhaps a model method?

Reply:
Sounds very doable. However let me ask this. I assuming this will be changed in the rewrite so, would the method become obsolete or is that better since the we will always need to display the old data?

Reply:
Good question. I guess it's a question of how we think it'll be implemented in the refactor, and whether that's something we can build towards now.

Reply:
I find logic in the view to be risky. If we kept it like this for now, I'd recommend that we rewrite it in the refactor anyway. I know @jforgue is working on this currently, and I'm glad.

Reply:
I've been trying to abstract this out, but it doesn't seem feasible to construct this in the controller or model. It seems the only way to not have logic in the view is by changing the options in the model before I pass them to the view. Which seems more texting and complex then a simple `|=`.

Reply:
What about something like this in the controller:

```ruby
@fields = BusStop::SUPER_HASH # we shouldn't be using model constants in the view anyway
@fields.each_pair do |category, fields|
  fields.each_pair do |field, options|
    if options.is_a?(Array) && options.exclude?(@stop.send(field))
      @fields[category][field] << @stop.send(field)
    end
  end
end
```